src/main/java/use_case/Snowflake.java

```diff
@@ -0,0 +1,95 @@
+package main.java.use_case;
+
+/**
+ * Revised from Twitter's snowflake algorithm for generating unique IDs
+ * Github: https://github.com/twitter-archive/snowflake
+ */
+public class Snowflake {
+    private long workerId;
+    private long datacenterId;
+    private long sequence;
+
+    private final long workerIdBits = 5L;  // 5 bits for worker id
+    private final long datacenterIdBits = 5L;  // 5 bits for data center id
+    private final long sequenceBits = 12L;  // 12 bits for serial number
+
+    // maximum values for the IDs
+    private final long maxWorkerId = ~(-1L << workerIdBits);
+    private final long maxDatacenterId = ~(-1L << datacenterIdBits);
+    private final long maxSequenceBits = ~(-1L << sequenceBits);
+
+    // bit shifting for each id
+    private final long workerIdShift = sequenceBits;
+    private final long dataCenterIdShift = sequenceBits + workerIdBits;
+    private final long timestampShift = sequenceBits + workerIdBits + datacenterIdBits;
+
+    private final long initialTimeStamp = 1288834974657L;
+    private long lastTimestamp = -1L;
+
+    public Snowflake(long workerId, long datacenterId, long sequence) {
+        if (workerId > maxWorkerId || workerId < 0) {
+            throw new IllegalArgumentException(
+                    String.format("worker ID cannot be greater than %d or less than 0", maxWorkerId));
+        } else if (datacenterId > maxDatacenterId || datacenterId < 0) {
+            throw new IllegalArgumentException(
+                    String.format("datacenter ID cannot be greater than %d or less than 0", maxDatacenterId));
+        }
+
+        this.workerId = workerId;
+        this.datacenterId = datacenterId;
+        this.sequence = sequence;
+    }
+
+    public long getWorkerId() {
+        return workerId;
+    }
+
+    public long getDatacenterId() {
+        return datacenterId;
+    }
+
+    public long getTimestamp() {
+        return System.currentTimeMillis();
+    }
+
+    /**
+     * generates the next ID
+     * @return the next ID
+     */
+    public synchronized long nextId() {
+        long timestamp = getTimestamp();
+
+        // check if clock has not moved backwards
+        if (timestamp < lastTimestamp) {
+            throw new RuntimeException(String.format(
+                    "Clock moved backwards. Refusing to generate id for %d milliseconds", lastTimestamp - timestamp));
+        }
+
+        // check if last timestamp equals the current timestamp
```

Comment:
irrelevant comment. The condition reads `lastTimestamp == timestamp` which says enough

Reply:
Same goes for the comment on lines 70, 77, and 81.
Comments should say important things that syntax, variable names, and method names do not. 
The comment on line 70 doesn't seem quite right. When we +1 and use the bitwise AND, we're not really increasing the sequence (in fact I think the bitwise & can only *decrease* the value, but I'm not 100% sure on that). What's the purpose of this operation and why are we using the bitwise AND? The answer to these kinds of things should be what comments entail.
The comment on 72 is ok, though I'd rather the purpose be the *reason* why we need to wait rather than just largely summarizing what the code pretty much already says.

Reply:
Oh also the line 16 comment is pretty redundant, too.

Reply:
Fixed